File: android/app/src/main/java/com/adqmobile/socialnetwork/domain/user/usecases/StoreUserUseCase.kt
package com.adqmobile.socialnetwork.domain.user.usecases

import com.adqmobile.socialnetwork.domain.user.User
import com.adqmobile.socialnetwork.domain.user.UserRepository

class StoreUserUseCase(private val userRepository: UserRepository) {

    suspend fun run(user: User) {
        userRepository.store(user)
    }
}

File: flutter/lib/domain/user/usecases/storeUserUseCase.dart
import '../userRepository.dart';

import '../user.dart';

class StoreUserUseCase {
  UserRepository userRepository;

  StoreUserUseCase(UserRepository userRepository) {
    this.userRepository = userRepository;
  }

  Future<User> run(User user) {
    return userRepository.store(user);
  }
}